<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QMainWindow</class>
 <widget class="QMainWindow" name="QMainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>850</width>
    <height>640</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>850</width>
    <height>640</height>
   </size>
  </property>
  <property name="font">
   <font>
    <pointsize>9</pointsize>
   </font>
  </property>
  <property name="windowTitle">
   <string>Homework 1 Demo 林季劼</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="dataset_text_label">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>20</y>
      <width>101</width>
      <height>27</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>標楷體</family>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="text">
     <string>資料集: </string>
    </property>
   </widget>
   <widget class="QPushButton" name="select_file_btn">
    <property name="geometry">
     <rect>
      <x>390</x>
      <y>20</y>
      <width>41</width>
      <height>27</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>標楷體</family>
      <pointsize>14</pointsize>
      <weight>50</weight>
      <bold>false</bold>
     </font>
    </property>
    <property name="text">
     <string>...</string>
    </property>
   </widget>
   <widget class="QLabel" name="selected_file_label">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>20</y>
      <width>211</width>
      <height>27</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Arial</family>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="styleSheet">
     <string notr="true">background: white;</string>
    </property>
    <property name="text">
     <string>No file selected</string>
    </property>
   </widget>
   <widget class="QWidget" name="figure_placeholder" native="true">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>100</y>
      <width>471</width>
      <height>421</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background: #cccccc;</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="normalized_cbox">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>57</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Agency FB</family>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Normalized</string>
    </property>
    <property name="iconSize">
     <size>
      <width>20</width>
      <height>20</height>
     </size>
    </property>
   </widget>
   <widget class="QSlider" name="pca_dim_slider">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>58</y>
      <width>160</width>
      <height>31</height>
     </rect>
    </property>
    <property name="minimum">
     <number>0</number>
    </property>
    <property name="maximum">
     <number>10</number>
    </property>
    <property name="value">
     <number>3</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
    <property name="invertedAppearance">
     <bool>false</bool>
    </property>
    <property name="invertedControls">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QLabel" name="cpa_label">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>57</y>
      <width>61</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Agency FB</family>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>PCA dim:</string>
    </property>
   </widget>
   <widget class="QLabel" name="show_pca_dim_label">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>57</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Agency FB</family>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">background:white;</string>
    </property>
    <property name="text">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;right&quot;&gt;3&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="show_train_data_cbox">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>560</y>
      <width>121</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Agency FB</family>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Show train Data</string>
    </property>
    <property name="iconSize">
     <size>
      <width>20</width>
      <height>20</height>
     </size>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QCheckBox" name="show_test_data_cbox">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>560</y>
      <width>111</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Agency FB</family>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Show test Data</string>
    </property>
    <property name="iconSize">
     <size>
      <width>20</width>
      <height>20</height>
     </size>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="learning_rate_spin_box">
    <property name="geometry">
     <rect>
      <x>600</x>
      <y>100</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="decimals">
     <number>4</number>
    </property>
    <property name="minimum">
     <double>0.000100000000000</double>
    </property>
    <property name="maximum">
     <double>10.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.000100000000000</double>
    </property>
    <property name="value">
     <double>0.001000000000000</double>
    </property>
   </widget>
   <widget class="QSpinBox" name="epochs_spin_box">
    <property name="geometry">
     <rect>
      <x>600</x>
      <y>140</y>
      <width>61</width>
      <height>22</height>
     </rect>
    </property>
    <property name="maximum">
     <number>10000</number>
    </property>
    <property name="value">
     <number>100</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="batchsize_spin_box">
    <property name="geometry">
     <rect>
      <x>600</x>
      <y>180</y>
      <width>61</width>
      <height>22</height>
     </rect>
    </property>
    <property name="value">
     <number>16</number>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>680</x>
      <y>100</y>
      <width>47</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>學習率</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>680</x>
      <y>140</y>
      <width>51</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>訓練次數</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>680</x>
      <y>180</y>
      <width>51</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>批次數量</string>
    </property>
   </widget>
   <widget class="QPushButton" name="start_train_btn">
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>220</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>訓練網路</string>
    </property>
   </widget>
   <widget class="QLabel" name="train_label">
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>350</y>
      <width>121</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微軟正黑體</family>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="text">
     <string>訓練準確率: </string>
    </property>
   </widget>
   <widget class="QLabel" name="test_label">
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>480</y>
      <width>121</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微軟正黑體</family>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="text">
     <string>測試準確率:</string>
    </property>
   </widget>
   <widget class="QLabel" name="loss_label">
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>310</y>
      <width>121</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微軟正黑體</family>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="text">
     <string>訓練誤差:</string>
    </property>
   </widget>
   <widget class="QPushButton" name="start_test_btn">
    <property name="geometry">
     <rect>
      <x>630</x>
      <y>430</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>測試網路</string>
    </property>
   </widget>
   <widget class="QLabel" name="show_loss_label">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>300</y>
      <width>171</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Arial</family>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="styleSheet">
     <string notr="true">background: white;</string>
    </property>
    <property name="text">
     <string>Not trained yet</string>
    </property>
   </widget>
   <widget class="QLabel" name="show_train_accuracy_label">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>340</y>
      <width>171</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Arial</family>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="styleSheet">
     <string notr="true">background: white;</string>
    </property>
    <property name="text">
     <string>Not trained yet</string>
    </property>
   </widget>
   <widget class="QLabel" name="show_test_accuracy_label">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>470</y>
      <width>171</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Arial</family>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="styleSheet">
     <string notr="true">background: white;</string>
    </property>
    <property name="text">
     <string>Not trained yet</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="show_lines_cbox">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>220</y>
      <width>141</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Agency FB</family>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Show Lines/Surface</string>
    </property>
    <property name="iconSize">
     <size>
      <width>20</width>
      <height>20</height>
     </size>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QCheckBox" name="show_wires_cbox">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>260</y>
      <width>171</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Agency FB</family>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Show Wireframe/Surface</string>
    </property>
    <property name="iconSize">
     <size>
      <width>20</width>
      <height>20</height>
     </size>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
